gef-ui/server/utils/helpers.js

```diff
@@ -52,10 +58,101 @@ const validationErrorHandler = (errs, href = '') => {
   };
 };
 
+const mapSummaryList = (data, itemsToShow) => {
+  if (!data || _isEmpty(data)) { return []; }
+  const { details, validation } = data;
+  const { required } = validation;
+
+  const valueObj = (val, isRequired, currency, options = {}) => {
+    if (isRequired && val === null) {
+      return { html: '<span class="has-text-danger" data-cy="required">Required</span>' };
+    }
+
+    if (val === null) {
+      return { text: '—' };
+    }
+
+    if (isObject(val) || Array.isArray(val)) {
+      const list = [];
+      Object.values(val).forEach((value) => {
+        if (value) {
+          list.push(`<li>${value}</li>`);
+        }
+      });
+
+      return { html: `<ul class="is-unstyled">${list.join('')}</ul>` };
+    }
+
+    if (options.isCurrency) {
+      return {
+        text: `${commaNumber(val)} ${currency}`,
```

Comment:
What's the requirement for currency formatting in GEF. On Portal, the requirement was for the format to be updated as the user type. As GEF is the same service I suspect the same functionality will be required.
see:
[https://github.com/foundry4/dtfs2/blob/1bc68911de1b7f5a189aeb434dc1293a83214c11/portal/templates/bond/bond-financial-details.njk#L35](https://github.com/foundry4/dtfs2/blob/1bc68911de1b7f5a189aeb434dc1293a83214c11/portal/templates/bond/bond-financial-details.njk#L35)

It uses iMask, which needs to be included in each page script. On page load it looks for the data-mask attribute
[https://github.com/foundry4/dtfs2/blob/1bc68911de1b7f5a189aeb434dc1293a83214c11/portal/scripts/masked-inputs.js#L20](https://github.com/foundry4/dtfs2/blob/1bc68911de1b7f5a189aeb434dc1293a83214c11/portal/scripts/masked-inputs.js#L20)

Reply:
The examples you have given are for inputs. Im just showing data (price with currency) in a summary list. When I start creating the inputs, I shall look at this.

Reply:
It also applies to text
[https://github.com/foundry4/dtfs2/blob/1bc68911de1b7f5a189aeb434dc1293a83214c11/portal/templates/bond/_macros/bond-submission-details.njk#L121](https://github.com/foundry4/dtfs2/blob/1bc68911de1b7f5a189aeb434dc1293a83214c11/portal/templates/bond/_macros/bond-submission-details.njk#L121)

Reply:
Ok.. will need to speak to you about this as Im not using any html as its generated by javascript.

Reply:
yeah, lets speak this afternoon. What happens when JS is switched off? For accessibility, it should still be rendered and working when JS is off.
[https://www.gov.uk/service-manual/technology/accessibility-for-developers-an-introduction](https://www.gov.uk/service-manual/technology/accessibility-for-developers-an-introduction)
[https://www.gov.uk/service-manual/technology/using-progressive-enhancement](https://www.gov.uk/service-manual/technology/using-progressive-enhancement)

Reply:
All the logic is server side, then to show it on the front end I use nunjucks for loop

`{% for item in facilities.data %}
    <h3 class="govuk-heading-m" data-cy="facility-type-heading">
      {{item.heading}} facility
    </h3>
    {{ govukSummaryList({
      rows: item.rows,
      attributes: {
        'data-cy': 'facility-summary-list'
      }
    }) }}
  {% endfor %}`

Reply:
Thats fine. I misunderstood and thought you were referring to client side JS.
I think you can add attributes to each item by populating the row.value.html property rather than row.value.text. 
[https://design-system.service.gov.uk/components/summary-list/#options-summary-list-example--rows](https://design-system.service.gov.uk/components/summary-list/#options-summary-list-example--rows)

So something like
``` 
html: `<span data-mask="currency">${commaNumber(val)} ${currency}</span>`,
```
It would ensure consistency of currency formatting across the site